File: data/src/main/java/org/oppia/data/backends/gae/model/ParamChange.java
package org.oppia.data.backends.gae.model;

import com.squareup.moshi.Json;

public class ParamChange {

  /*
   * Parent Hierarchy: ExplorationContainer -> Exploration -> ParamChange
   * Parent Hierarchy: ExplorationContainer -> Exploration -> State -> ParamChange
   */

  @Json(name = "generator_id")
  public String generatorId;

  @Json(name = "name")
  public String name;

  @Json(name = "customization_args")
  public CustomizationArgs customizationArgs;

}

File: data/src/main/java/org/oppia/data/backends/gae/model/ParamChange.kt
package org.oppia.data.backends.gae.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class ParamChange(

  @Json(name = "generator_id") val generatorId: String?,
  @Json(name = "name") val name: String?,
  @Json(name = "customization_args") val customizationArgs: CustomizationArgs?

)
